src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -1778,6 +1864,23 @@ def _configure_security_policy_storage_params(arg_ctx):
                    options_list=['--maint-config-id', '-m'],
                    help='Assign maintenance configuration to this managed instance.')
 
+        c.argument('enable_ad_only_auth',
+                   options_list=['--enable-ad-only-auth'],
+                   help='Enable Azure Active Directory Only Authentication for this server.')
+
+        c.argument('external_admin_name',
+                   options_list=['--external-admin-name'],
+                   help='Display name of the Azure AD administrator user, group or application.')
+
+        c.argument('external_admin_sid',
+                   options_list=['--external-admin-sid'],
+                   help='The unique ID of the Azure AD administrator. Object Id for User or Group, '
+                   'Client Id for Applications')
+
+        c.argument('external_admin_principal_type',
+                   options_list=['--external-admin-principal-type'],
+                   help='User, Group or Application')
```

Comment:
what about use get_enum_type() for allowed values

Reply:
I have tested all my scenarios with this, no bandwidth to run server and mi tests again